qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -0,0 +1,304 @@
+                new_coeff *= (-1) ** ncommutations
+
+            # Check what happens to the symbol
+            new_char = mapping[char1 + char2]
+            if new_char == 0:
+                return "I" * len(label1), 0
+            new_label += cast(str, new_char)
+
+        return new_label, new_coeff
+
+    def __add__(self, other):
+        """Returns a `FermionicOp` representing the sum of the given base fermionic
+        operators.
+        """
+        if not isinstance(other, FermionicOp):
+            raise TypeError(
+                "Unsupported operand type(s) for +: 'FermionicOp' and "
+                "'{}'".format(type(other).__name__)
+            )
+
+        # Check compatibility (i.e. operators act on same register length)
+        assert self._is_compatible(other), "Incompatible register lengths for '+'. "
+
+        label1, coeffs1 = zip(*self.to_list())
+        label2, coeffs2 = zip(*other.to_list())
+
+        return FermionicOp(list(zip(label1 + label2, coeffs1 + coeffs2))).reduce()
+
+    def to_list(self) -> List[Tuple[str, complex]]:
+        """Getter for the operator_list of `self`"""
+        return list(zip(self._labels, self._coeffs))
+
+    @property
+    def register_length(self):
+        """Getter for the length of the fermionic register that the FermionicOp `self` acts
+        on.
+        """
+        return self._register_length
+
+    def dagger(self):
+        """Returns the complex conjugate transpose (dagger) of `self`."""
+
+        dagger_map = {"+": "-", "-": "+", "I": "I", "N": "N", "E": "E"}
+        label_list = []
+        coeff_list = []
+        for label, coeff in zip(self._labels, self._coeffs):
+            conjugated_coeff = coeff.conjugate()
+
+            daggered_label = ""
+
+            for i, char in enumerate(label):
+                daggered_label += dagger_map[char]
+                if char in ["+", "-"]:
+                    permute_through = label[i + 1:]
+                    conjugated_coeff *= (-1) ** (
+                        permute_through.count("+") + permute_through.count("-")
+                    )
+
+            label_list.append(daggered_label)
+            coeff_list.append(conjugated_coeff)
+
+        return FermionicOp(list(zip(label_list, coeff_list)))
+
+    def _is_compatible(self, operator) -> bool:
+        """
+        Checks whether the `operator` is compatible (same shape and
+
+        Args:
+            operator (FermionicOperator/FermionicOp): a fermionic operator
+
+        Returns:
+            True iff `operator` is compatible with `self`.
+        """
+        return (
+            isinstance(operator, FermionicOp)
+            and self.register_length == operator.register_length
+        )
+
+    def reduce(self) -> "FermionicOp":
+        """
+        Reduce
+
+        Returns:
+            The reduced `FermionicOp`
+        """
+        # TODO: atol, rtol
+        label_list, indexes = np.unique(self._labels, return_inverse=True, axis=0)
```

Comment:
I think add operation is commutative since fermion follows CAR algebra.

Reply:
@ikkoham you are right. The sum ordering is not important. The only point I see in @mrossinek comment is that preserving the order will be possibly easier to read the operator later on.

Reply:
That is only the case if the operator is constructed in some meaningful order. Otherwise the sorted list may actually be easier to read later.. I don't have a strong opinion on whether we need to preserve the order (since mathematically it makes no difference). I don't think it causes much overhead to keep it sorted and it is probably what the end-user expects. So personally, I would sort it again, but we can also leave it be.

Reply:
OK. `Reduce()` is costly, so don't do it inside add. When users want to reduce, use reduce() explicitly.

Reply:
This can be resolved since you removed `reduce()`, right?

Reply:
Yes!